From: Jiri Slaby <jslaby@suse.cz>
Date: Tue, 2 Jul 2019 12:34:11 +0200
Subject: Revert "ovl: fix bogus -Wmaybe-unitialized warning"
Patch-mainline: never, suse specific
References: bnc#1140057

This reverts commit 9b8bd98e23a92b4fa6b1f3dd9e381c47ce575642, upstream
commit 1dac6f5b0ed2601be21bb4e27a44b0c3e667b7f4. It depends on the
following patch which is being reverted.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 fs/overlayfs/super.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/overlayfs/super.c
+++ b/fs/overlayfs/super.c
@@ -996,8 +996,8 @@ static int ovl_setup_trap(struct super_b
 	int err;
 
 	trap = ovl_get_trap_inode(sb, dir);
-	err = PTR_ERR_OR_ZERO(trap);
-	if (err) {
+	err = PTR_ERR(trap);
+	if (IS_ERR(trap)) {
 		if (err == -ELOOP)
 			pr_err("overlayfs: conflicting %s path\n", name);
 		return err;
